### PR TITLE
Task change assignee

### LINKED ISSUE #77: Add change assignee function for Task model

### PR DESCRIPTION
**Pull request by orzionpour**
Fixes #77 
This PR adds the ability to change the assigned employee of a given task in DB.

## Changes you made

**- Create conftest.py for reusable fixtures** 
Added fixtures that create a test DB we can use in future tests (and refactor previous tests to prevent code duplications that we already have).
**- Add change_assignee function to Task model**
This function is a Task instance's function that changes the assigned employee for the given task.
**- Add change_assignee tests**
Tests for change_assignee tests.

### CODE REVIEW

**Review by Asafh7 — APPROVED** [review]

**Review by AvivLiberman — APPROVED** [review]

**Comment by nunnatsa** [comment]
@orzionpour - Please add meaningful description. I human language - what is the change done in this PR - not which files are added etc. but something that your readers (including your future self) will understand.

File: conftest.py
Comment (nunnatsa) [inline conftest.py:33]: please use named parameters.

File: conftest.py
Comment (nunnatsa) [inline conftest.py:38]: please use named parameters.

File: conftest.py
Comment (nunnatsa) [inline conftest.py:34]: I wouldn't use random input in unit test. Random test are important, but unit test should be more robust and predictable. This is the first layer of code reliability and I think that random tests are fit for this purpose.

**Review by nunnatsa — COMMENTED** [review]
Added some non-blocking comment. Please consider to fix them.

**Comment by orzionpour** [comment]
> @orzionpour - Please add a meaningful description, In a human language - what is the change done in this PR - not which files are added etc. but something that your readers (including your future self) will understand.

Done.

File: tasks/models.py
Comment (nunnatsa) [inline tasks/models.py:57]: What will happen if the new_assignee does not exist?

**Review by Asafh7 — APPROVED** [review]

File: tasks/models.py
Comment (orzionpour) [reply tasks/models.py:57]: Good point. Made some changes. Uploading the PR right now.
Anyway, there are some cases that Django raises an exception, and we can use it.

**Review by aradz1997 — APPROVED** [review]

File: tasks/models.py
Comment (nunnatsa) [reply tasks/models.py:62]: I wouldn't prevent task to move from team to team, but this is a business decision and I'll leave it to you.

But what if the new_assignee is not in the DB? Not part of the system?

Comment (orzionpour) [reply tasks/models.py:62]: Is it possible? I mean when using the create_user or User.objects.create the user is saved in the DB.

**Review by Asafh7 — APPROVED** [review]

File: tasks/models.py
Comment (orzionpour) [reply tasks/models.py:62]: The implementation is simple here. But as I understand, once I create a user, it is saved in DB.
So I'm not sure how to write a test for this one

Comment (AvivLiberman) [reply tasks/models.py:64]: @nunnatsa You can't assign a task to someone who is not exist, when the manager will edit the task he could only chose members of his team (which must be existing)

Comment (orzionpour) [reply tasks/models.py:64]: Resolved with the last commit.

File: conftest.py
Comment (kobihk) [inline conftest.py:38]: you can use the append method
```suggestion
            employees.append(employee)
```

File: conftest.py
Comment (kobihk) [inline conftest.py:48]:
```suggestion
        managers.append(manager)
```

**Review by kobihk — CHANGES_REQUESTED** [review]

File: conftest.py
Comment (orzionpour) [reply conftest.py:29]: I used `for i in range(3)` for creating 3 users per team.
`users_counter` was used to resolve issues with non-unique usernames/emails that I encountered during their instantiation. Note that `users_counter` is raised by 1 on every iteration inside the loop (in this case, up to 9, while `i` up to 3)

**Review by avraham1992 — DISMISSED** [review]

File: conftest.py
Comment (kobihk) [reply conftest.py:26]: OK. Thanks for the detailed explanation.

**Comment by orzionpour** [comment]
Resolved everything.
Note that there will be conflicts when merging #74 and #83, so please after merging one of them, wait so I will resolve the conflicts (the conflicts are only because I worked on the same model file)

File: conftest.py
Comment (kobihk) [inline conftest.py:58]: I would prefer to avoid one letter variable and to give a meaningful name.
IMHO this code is less readable and needs to dig into the code to understand for what we have each loop and for what k,I,j stand for

File: conftest.py
Comment (kobihk) [inline conftest.py:79]: looks the same as:
```suggestion
    return tuple(tasks[:4])
```

File: tasks/tests/test_change_assignee.py
Comment (kobihk) [inline tasks/tests/test_change_assignee.py:29]: from the next two lines, it looks better to check it like:
```suggestion
        assert len(teams) >= 2
```
because you need teams[0] and teams[1]

File: tasks/tests/test_change_assignee.py
Comment (kobihk) [inline tasks/tests/test_change_assignee.py:52]: maybe you can do it like:
```suggestion
        task = test_db[3][0]
```
as you do it in the previous test in line # 41

**Review by kobihk — CHANGES_REQUESTED** [review]

File: conftest.py
Comment (orzionpour) [reply conftest.py:30]: Resolved.

File: conftest.py
Comment (orzionpour) [reply conftest.py:14]: Resolved

File: conftest.py
Comment (orzionpour) [reply conftest.py:45]: Resolved

File: conftest.py
Comment (orzionpour) [reply conftest.py:58]: Resolved

File: conftest.py
Comment (orzionpour) [reply conftest.py:79]: The explicit tuple components here were intentioned, to make it more readable to those who want to use this fixture.

File: tasks/tests/test_change_assignee.py
Comment (orzionpour) [reply tasks/tests/test_change_assignee.py:52]: Resolved

File: tasks/tests/test_change_assignee.py
Comment (orzionpour) [reply tasks/tests/test_change_assignee.py:29]: Resolved.

File: conftest.py
Comment (kobihk) [reply conftest.py:85]: OK, thanks!!

**Review by kobihk — APPROVED** [review]

**Review by Asafh7 — APPROVED** [review]

**Comment by orzionpour** [comment]
Resolved conflicts with #74 - can be merged now.

**Review by kobihk — APPROVED** [review]

**Review by AvivLiberman — APPROVED** [review]

**Review by aradz1997 — APPROVED** [review]

**Review by avraham1992 — APPROVED** [review]

**Pull request merged**
